### PR TITLE
[core] Change the API to support rediss://

### PR DESCRIPTION
**Pull request by fishbone**
## Why are these changes needed?
The current API is to use RAY_ENABLE_REDIS_SSL as the os env. The updated API asks to pass rediss:// as the redis address.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by iycheng** [comment]
@simon-mo I thought I added it in my last PR, and it turned out not...

File: bazel/BUILD.redis [hunk 1]
```diff
@@ -48,13 +48,8 @@ genrule(
         "redis-cli",
     ],
     cmd = """
-    mkdir tmp-redis-bin
-    cp -rf $(locations :redis) ./tmp-redis-bin/
-    cp tmp-redis-bin/redis-server $(location redis-server)
-    chmod +x $(location redis-server)
-    cp tmp-redis-bin/redis-cli $(location redis-cli)
-    chmod +x $(location redis-cli)
-    rm -rf tmp-redis-bin
```
Comment (simon-mo) [inline bazel/BUILD.redis:57]: is this change no longer necessary?

File: python/ray/_private/services.py [hunk 2]
```diff
@@ -1089,6 +1090,7 @@ def _start_redis_instance(
         listen_to_localhost_only: Redis server only listens to
             localhost (127.0.0.1) if it's true,
             otherwise it listens to all network interfaces.
+        enable_redis: Enable the TLS/SSL in Redis or not
```
Comment (simon-mo) [inline python/ray/_private/services.py:1093]:
```suggestion
        enable_tls: Enable the TLS/SSL in Redis or not
```

**Review by simon-mo — APPROVED** [review]

File: bazel/BUILD.redis (hunk 1 above)
Comment (iycheng) [reply bazel/BUILD.redis:57]: This is in this PR https://github.com/ray-project/ray/pull/29179
Otherwise, I can't rebuild it locally...

Comment (iycheng) [reply bazel/BUILD.redis:57]: Let me merge to the upstream

**Comment by iycheng** [comment]
it seems the minimal build is failing in the master: https://buildkite.com/ray-project/oss-ci-build-branch/builds/430#0183b4b0-28aa-4c4d-a9a4-3642713f6fba

**Pull request merged**